cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -152,6 +152,10 @@ struct ARROW_EXPORT S3Options {
 
   S3Options();
 
+  /// Get a stock AWS retry strategy from a string.
+  static std::shared_ptr<S3RetryStrategy> GetS3RetryStrategy(const std::string& name,
+                                                             long retry_attempts);
```

Comment:
Can you make this `int64_t max_attempts`? (i.e. use `int64_t` instead of `long`, and name the argument better)